### DISCUSSION TITLE
Access fail "401: Unauthorized" downloading entries from Nighscout in Google Cloud with HTTP

### DISCUSSION THREAD

**Discussion by Brunildogithub** [NightscoutFoundation/xDrip#2437] [Q&A] [answered]
Nightscout in Google Cloud running correctly, master uploading data successfully. But failed in following with xDrip.
Under “Nightscout Follow URL”, I enter http://Token@hostname assuming Token is the token from the master, generated as explained in the instructions, and hostname is my noip.com hostname. But No data downloaded, I get "NS entries download was not successful:  401: Unauthorized".

If I try the following in a browser on the follower phone: http://hostname/?token=Token, where hostname is my noip.com hostname and Token is the token from the master, it works correctly. But no success in Xdrip follower using the same url.

I guess the problem may be using http instead of https. But due to a firewall issue, I had to allow all in the VM so now I am forced to use http to access hostname instead of https.

http://hostname/?token=Token works successfully in a browser on the follower phone, but no in Xdrip Follower.

Please advice.

**Comment by Navid200** [comment, marked as answer]
@jamorham @tzachi-dar Guys this firewall issue needs to be addressed.  
Why are people claiming that there is anything wrong with the firewall.
None of us has had any problems with the firewall.

What can I do to help people who say they have problems with firewall?

**Comment by Navid200** [comment]
@Brunildogithub Are you willing to help me figure out what the problem is?
Or, do you know someone who may be?

Would you please use a different Google account and exactly follow the following instructions to create a new Nightscout on Google Cloud?
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud

When you get to the point that something doesn't work, please take snapshots and show me.

I would like to see exactly what the problem is that has made some people modify the firewall.

**Reply by Brunildogithub** [reply]
Of course Navid, I'll try to do so. Thanks a lot for your help. I'll return to you with feedback asap.

**Comment by tzachi-dar** [comment]
Please make sure to mark the following when creating the machine:![image](https://user-images.githubusercontent.com/7942372/200319765-479f22ec-88eb-45d4-82cb-12cae3186d42.png)

**Reply by Navid200** [reply]
That's in the instructions already.  But, I just circled those in red to add emphasis.  
https://navid200.github.io/xDrip/docs/Nightscout/NS_FreeTier

**Comment by Brunildogithub** [comment]
I'm confused. When checking VM, I see Firewall traffic both HTTP and HTTPS inactive. But when I edit, I see both Traffic allowed. Snapshots hereunder:

![image](https://user-images.githubusercontent.com/38180297/200327759-54f43d3a-895c-42e0-9b3f-ead356b5d332.png)

![image](https://user-images.githubusercontent.com/38180297/200328184-718bd357-a984-46ea-8ce6-5a3aedebc494.png)

Not sure how to grasp this. Please advice.

**Reply by Navid200** [reply]
That's weird.  I just created a new machine, enabled those two and after creating, it shows on for both.

How did you create this following the guide or using an image you downloaded from somewhere?

**Reply by Brunildogithub** [reply]
I followed a guide from a spanish collaborator. My thought was that this guide was the same as yours. But indeed he included an automatism to create the VM through Cloud Shell console and I'm afraid there is some bug in the automatism that results in a firewall issue.
Is there any chance to fix the issue? What should I do to get Firewall traffic ON?
Or should I delete my VM and start from scratch with your guide?

**Reply by Navid200** [reply]
You should backup your MongoDB, and download it to your computer.
You should also download your nsconfig file containing your Nightscout variables.

Then, you can delete the machine and create a new one following our guide.
Then, you can upload your nsconfig file and your MongoDB backup file.

Then, you replace the existing nsconfig file on the new machine with the one you downloaded.  That updates all your Nightscout variables to exactly what you have now.

Then, you use MongoDB restore utility to restore the MongoDB database you backed up.
Then, you will have everything back up and running minus the activity during this procedure.
For that, you can use the xDrip backfill utility to upload  the missing data to your new Nightscout.

@tzachi-dar Do you approve?

**Reply by tzachi-dar** [reply]
Well, what Navid has suggested is the way to get you settled with an "approved" installation.

**Reply by Brunildogithub** [reply]
Thanks a lot Navid and tzachi-dar for your unvaluable help. I will follow your advice in order to set a new installation.

**Comment by EasternPA** [comment]
I'm running GrapheneOS and I'm getting this 401 error. I'm using CloudFlare Tunnels, and I have SSL working. I've tried the URL/?token= format with both the API token configured in docker and the API key generated in the Nightscout web UI. I initially tried to use the xDrip+ Sync Follower but it failed because I don't have Google Play Services installed. I am no longer getting the Google Play Services error, but 401 instead.

What is the proper URL format for the Nightscout follower and which token am I supposed to use with the URL?

Thanks.

Edit: Never mind. I got it to work with "https://< token from docker container env variable >@url/"

**Reply by Navid200** [reply]
I can help only if you follow our guide:
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html 

If you do and create a setup, the first step you are asked to complete is to confirm that you can access your site in a web browser.
Have you done that?

The formats related to xDrip as an uploader are explained here:  https://navid200.github.io/xDrip/docs/Nightscout-Settings.html 
If you are using xDrip as a follower, the steps are explained here: https://navid200.github.io/xDrip/docs/Follow/FollowNightscout.html
The token is explained here: https://navid200.github.io/xDrip/docs/Nightscout/Tokens.html 

Have a look at those and let me know if you still have any questions.

**Reply by EasternPA** [reply]
Thank you, but it is self-hosted, not a Cloud instance, so it was unrelated to the ports and protocols being allowed. The format of the follower URL is

https://< token >@fqdn

That was the answer I was looking for.
